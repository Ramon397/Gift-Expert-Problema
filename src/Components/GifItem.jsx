import PropTypes from 'prop-types';

export const GifItem = ({ title, url, id }) => {

  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,  
  url: PropTypes.string.isRequired
}


/*
1.-a√±adir protype yarsn add
a.title obligatorio
b.url obligatotio


2.-evaluar el sabapshot



*/
