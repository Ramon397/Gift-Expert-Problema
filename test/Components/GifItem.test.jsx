import { render } from "@testing-library/react";
import { GifItem } from "../../src/Components/GifItem";
//Declaracion de constantes
const title='Saitana';
const url='http://www.google.com';

//Prueba de  funcionalidades

describe('Pruebas en < GifItem />', () => { 
//Primer Test   
    test('debe de hacer macch con el snapshot', () => { 
       const {container} =render(<GifItem title={title}  url={url}/>);
       expect(container).toMatchSnapshot();
     });
//Segundo test 
     test('Debe de mostrar la imagen  con el url y el Alt  indicado', () => { 
        render(<GifItem title={title} url={url}/>)
        screen.Debug();
     });
//Tercer Test
    // test('Debe de mostrar el titulo en el componente', () => { 
    //     render(<GifItem title={title}  url={url}/>);
    //     expect(screen.getbyText(title)).toBeTruthy();
    // });

});