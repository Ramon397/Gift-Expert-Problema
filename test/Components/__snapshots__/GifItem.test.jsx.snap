// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en < GifItem /> debe de hacer macch con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitana"
      src="http://www.google.com"
    />
    <p>
      Saitana
    </p>
  </div>
</div>
`;
